server/php-engine.cpp

```diff
@@ -1545,25 +1554,23 @@ int rpcx_execute(connection *c, int op, raw_message *raw) {
       if (in_sigterm) {
         return 0;
       }
-      // got a new task from the tasks engine
+      // got a new task from the tasks engine or a request from RPC microservice client
       tl_fetch_init_raw_message(raw);
-      auto op_from_tl = tl_fetch_int();
-      len -= static_cast<int>(sizeof(op_from_tl));
-      assert(op_from_tl == op);
-      assert(len % sizeof(int) == 0);
 
-      auto req_id = tl_fetch_long();
-      len -= static_cast<int>(sizeof(req_id));
+      int64_t left_bytes_with_headers = tl_fetch_unread();
+      tl_query_header_t header{};
+      assert(tl_fetch_query_header(&header, false) && "Can't fetch RPC query headers, terminate worker");
```

Comment:
1. fixed
2. At first I tried to send RPC error instead of terminating worker, but the problem is that rpc-proxy expects correct `req_id` which is the part of the header... But I think we can assume that `req_id` is almost always fetched correctly and do this anyway. Changed this place